service/src/main/java/tds/assessment/repositories/impl/AssessmentCommandRepositoryImpl.java

```diff
@@ -0,0 +1,140 @@
+ * Community License, Version 2.0 (the “license”); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the license at
+ *
+ * https://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required under applicable law or agreed to in writing, software distributed under the
+ * License is distributed in an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for specific language governing permissions
+ * and limitations under the license.
+ **************************************************************************************************/
+
+package tds.assessment.repositories.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import tds.assessment.Assessment;
+import tds.assessment.repositories.AssessmentCommandRepository;
+
+@Repository
+public class AssessmentCommandRepositoryImpl implements AssessmentCommandRepository {
+    private final JdbcTemplate jdbcTemplate;
+    private static final String[] ITEMBANK_TABLE_NAMES = {
+        "aa_itemcl",
+        "tblsetofitemstrands",
+        "tblsetofitemstimuli",
+        "tblitemprops",
+        "setoftestgrades",
+        "testcohort",
+        "tblitemselectionparm",
+        "tbladminstrand",
+        "tblsetofadminitems",
+        "itemscoredimension",
+        "tbladminstimulus",
+        "testform",
+        "testformitem",
+        "affinitygroup",
+        "affinitygroupitem",
+        "tblsetofadminsubjects"
+    };
+
+    private static final String[] CONFIGS_TABLE_NAMES = {
+        "client_testmode",
+        "client_testformproperties",
+        "client_testwindow",
+        "client_testgrades",
+        "client_testeligibility",
+        "client_test_itemtypes",
+        "client_test_itemconstraint",
+        "client_testtooltype",
+        "client_segmentproperties",
+        "client_testproperties"
+    };
+
+    @Autowired
+    public AssessmentCommandRepositoryImpl(final JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public void removeAssessmentData(final String clientName, final Assessment assessment) {
+        final List<String> SQL = generateItembankDeleteSQL(assessment.getKey());
+        SQL.addAll(generateConfigsDeleteSQL(clientName, assessment.getAssessmentId()));
+
+        jdbcTemplate.batchUpdate(SQL.toArray(new String[SQL.size()]));
+    }
+
+    private List<String> generateItembankDeleteSQL(final String assessmentKey) {
+        final List<String> SQL = Arrays.stream(ITEMBANK_TABLE_NAMES)
+            .map(table -> table.equals("tblsetofadminsubjects")
+                ? String.format("DELETE FROM itembank.%s WHERE _key = '%s';", table, assessmentKey)
```

Comment:
Is there not a way to use one of the other batch update methods that allow one to use prepared statements rather than building these queries up by hand?

Reply:
There is not unfortunately. The `NamedParameterJdbcTemplate` does not provide a method that allows multiple separate statemend with `batchUpdate()`. @malaffoon Attempted to find a way around this and ran into a number of issues.